File: static/CWE-338/javascript/patched.js
// Remediation for CWE-338: Use of Cryptographically Weak Pseudo-Random Number Generator (PRNG)
// Fix: Generate the reset token with crypto.randomBytes instead of the predictable Math.random().

const express = require('express');
const crypto = require('crypto');

const app = express();

function generateResetToken() {
  return crypto.randomBytes(32).toString('hex');
}

app.post('/reset-password', (req, res) => {
  const token = generateResetToken();
  res.json({ resetToken: token });
});

app.listen(3000);

File: static/CWE-330/python/patched.py
# Remediation for CWE-330: Use of Insufficiently Random Values
# Fix: Generate tokens with the cryptographically secure secrets module.

import secrets


def generate_reset_token(user_id):
    token = "".join(str(secrets.randbelow(10)) for _ in range(16))
    nonce = secrets.token_hex(16)
    return f"{user_id}-{token}-{nonce}"
